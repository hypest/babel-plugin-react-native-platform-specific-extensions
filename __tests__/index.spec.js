import pluginTester from "babel-plugin-tester";
import platformSpecific from "../index";
import fs from "fs";

let spy;

pluginTester({
  plugin: platformSpecific,
  pluginName: "babel-plugin-react-native-platform-specific-extensions",
  pluginOptions: {
    extensions: ["scss"]
  },
  snapshot: true,
  tests: [
    {
      title: "Should require ios and android files if they exits",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /styles\.ios\.scss/.test(path) || /styles\.android\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should require ios and native files if they exits",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /styles\.ios\.scss/.test(path) || /styles\.native\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should require android and native files if they exits",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /styles\.android\.scss/.test(path) ||
            /styles\.native\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should require ios and non prefixed file",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return /styles\.ios\.scss/.test(path);
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should require android and non prefixed file",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return /styles\.android\.scss/.test(path);
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should require native file if it exists",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return /styles\.native\.scss/.test(path);
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should preserve import name and filename transforming names",
      code: `import myStyles from "./myStylesFile.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /myStylesFile\.ios\.scss/.test(path) ||
            /myStylesFile\.android\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should not touch import if the files do not exist",
      code: `import styles from "./styles.scss"`
    },
    {
      title: "Should not do anything with a global import (to be implemented)",
      code: `import "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /styles\.ios\.scss/.test(path) || /styles\.android\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    }
  ]
});

pluginTester({
  plugin: platformSpecific,
  pluginName: "babel-plugin-react-native-platform-specific-extensions",
  pluginOptions: {
    extensions: ["txt", "json"]
  },
  snapshot: true,
  tests: [
    {
      title: "Should work with other extension types (.txt)",
      code: `import txt from "./something.txt"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /something\.ios\.txt/.test(path) ||
            /something\.android\.txt/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should work with other extension types (.json)",
      code: `import json from "./something.json"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /something\.ios\.json/.test(path) ||
            /something\.android\.json/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title:
        "Should not react to file types that are not defined in plugin options (.scss)",
      code: `import styles from "./styles.scss"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /styles\.ios\.scss/.test(path) || /styles\.android\.scss/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    },
    {
      title: "Should not do anything with a global import (to be implemented)",
      code: `import "./something.txt"`,
      setup() {
        spy = jest.spyOn(fs, "existsSync").mockImplementation(path => {
          return (
            /something\.ios\.txt/.test(path) ||
            /something\.android\.txt/.test(path)
          );
        });
      },
      teardown() {
        spy.mockRestore();
      }
    }
  ]
});
