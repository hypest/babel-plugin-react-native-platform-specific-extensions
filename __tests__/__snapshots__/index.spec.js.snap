// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-native-platform-specific-extensions Should not do anything with a global import (to be implemented): Should not do anything with a global import (to be implemented) 1`] = `
"
import \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import \\"./styles.scss\\";
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should not do anything with a global import (to be implemented): Should not do anything with a global import (to be implemented) 2`] = `
"
import \\"./something.txt\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import \\"./something.txt\\";
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should not react to file types that are not defined in plugin options (.scss): Should not react to file types that are not defined in plugin options (.scss) 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import styles from \\"./styles.scss\\";
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should not touch import if the files do not exist: Should not touch import if the files do not exist 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import styles from \\"./styles.scss\\";
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should preserve import name and filename transforming names: Should preserve import name and filename transforming names 1`] = `
"
import myStyles from \\"./myStylesFile.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var myStyles = Platform.OS === \\"ios\\" ? require(\\"./myStylesFile.ios.scss\\") : require(\\"./myStylesFile.android.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require android and native files if they exits: Should require android and native files if they exits 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var styles = Platform.OS === \\"android\\" ? require(\\"./styles.android.scss\\") : require(\\"./styles.native.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require android and non prefixed file: Should require android and non prefixed file 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var styles = Platform.OS === \\"android\\" ? require(\\"./styles.android.scss\\") : require(\\"./styles.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require ios and android files if they exits: Should require ios and android files if they exits 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var styles = Platform.OS === \\"ios\\" ? require(\\"./styles.ios.scss\\") : require(\\"./styles.android.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require ios and native files if they exits: Should require ios and native files if they exits 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var styles = Platform.OS === \\"ios\\" ? require(\\"./styles.ios.scss\\") : require(\\"./styles.native.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require ios and non prefixed file: Should require ios and non prefixed file 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var styles = Platform.OS === \\"ios\\" ? require(\\"./styles.ios.scss\\") : require(\\"./styles.scss\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should require native file if it exists: Should require native file if it exists 1`] = `
"
import styles from \\"./styles.scss\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import styles from \\"./styles.native.scss\\";
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should work with other extension types (.json): Should work with other extension types (.json) 1`] = `
"
import json from \\"./something.json\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var json = Platform.OS === \\"ios\\" ? require(\\"./something.ios.json\\") : require(\\"./something.android.json\\");
"
`;

exports[`babel-plugin-react-native-platform-specific-extensions Should work with other extension types (.txt): Should work with other extension types (.txt) 1`] = `
"
import txt from \\"./something.txt\\"

      ↓ ↓ ↓ ↓ ↓ ↓

import { Platform } from \\"react-native\\";
var txt = Platform.OS === \\"ios\\" ? require(\\"./something.ios.txt\\") : require(\\"./something.android.txt\\");
"
`;
